client/services/FeedbackGeneratorService.js

```diff
@@ -67,7 +68,7 @@ tie.factory('FeedbackGeneratorService', [
             "more efficiently?"].join(''));
           return feedback;
         }
-        if (errorMessage) {
+        else if (errorMessage) {
```

Comment:
One small style nit: can you move this to the previous line? So:

`} else if (errorMessage) {`

That way, it's clearer it's part of the same block. Eventually this will be in a lint check too, but thought I'd mention it here so that the alternative pattern doesn't propagate.